### PR TITLE
Add code intel extensions bundles

### LINKED ISSUE #37550: Extensions: create repo with code-intel extensions bundles
In order to minimize the admin's effort of enabling code intelligence on the air-gapped instances create a repo with code intel extension bundles and manifests.
Except for the latter two, this repo should also include the [code-intel-extensions](https://github.com/sourcegraph/code-intel-extensions) revision from which the bundles were created, so that admins can run security audits considering this revision. 
More details in [this thread](https://sourcegraph.slack.com/archives/C01LZKLRF0C/p1655302657310299).

### PR DESCRIPTION
**Pull request by taras-yemets**
Closes https://github.com/sourcegraph/sourcegraph/issues/37550

_**Note:**_ extension manifests and bundles are generated and don't need to be reviewed. Their generation and publishing logic lives inside `index.js` and `bundles/publish.js` respectively and should be reviewed.

### CODE REVIEW

**Review by vdavid — APPROVED** [review]
STAMP

File: README.md [hunk 1]
```diff
@@ -1 +1,22 @@
...
+     |Environment Variable|Description|
+     |--------------------|-----------|
+     | `SRC_ENDPOINT` | See `src-cli`'s [README](https://github.com/sourcegraph/src-cli#log-into-your-sourcegraph-instance) for more information. |
+     | `SRC_ACCESS_TOKEN` | See `src-cli`'s [README](https://github.com/sourcegraph/src-cli#log-into-your-sourcegraph-instance) for more information. |
+     | `PUBLISHER` | The name of the [publisher](https://docs.sourcegraph.com/extensions/authoring/manifest#fields) |
+
+   - NOTE: You do _not_ need to have `src-cli` installed for this script to work.
+
+#### Update the Extension Bundles
+1. Update the `revision.txt` file with the target revision of [code-intel-extensions repo](https://github.com/sourcegraph/code-intel-extensions]).
```
Comment (abeatrix) [inline README.md]:
```suggestion
1. Update the `revision.txt` file with the target revision of [code-intel-extensions repo](https://github.com/sourcegraph/code-intel-extensions).
```

File: index.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
+const fs = require("fs");
+const path = require("path");
+const util = require("node:util");
+
+const { revision } = require("./config");
```
Comment (abeatrix) [inline index.js:5]: Maybe we can get the latest revision number automatically via git command? 

```
git ls-remote https://github.com/sourcegraph/code-intel-extensions HEAD
```

**Review by abeatrix — APPROVED** [review]
LGTM! Confirmed the scripts run for me:
image
image



I wonder if it would be easier if we move the scripts to the [code-intel-extensions repo](https://github.com/sourcegraph/code-intel-extensions). The user will need to generate the files locally but that way they will always have the latest version of the extensions, thoughts?

**Comment by taras-yemets** [comment]
Thank you for your review, @abeatrix!

>I wonder if it would be easier if we move the scripts to the [code-intel-extensions repo](https://github.com/sourcegraph/code-intel-extensions). The user will need to generate the files locally but that way they will always have the latest version of the extensions, thoughts?

There is a note in the [original request](https://sourcegraph.slack.com/archives/C01LZKLRF0C/p1655302688359379?thread_ts=1655302657.310299&cid=C01LZKLRF0C) for this change that the customers’ scanning process is typically fully automated as long as they can provide a public URL to what is being scanned. So for customers to scan the extension code, they need to have a bundle URL (bundles should be added to the repo). In the `code-intel-extensions` repo, extensions are generated and published [in CI](https://github.com/sourcegraph/code-intel-extensions/blob/master/.buildkite/gen-pipeline.sh#L38) after the related PR is merged to the main branch. To implement it, we probably need to add another step to the CI, which will create a PR with the generated bundles after they are published. Given that we don't expect to generate code-intel extensions to be used on the air-gapped instances that often (usually they are added once and rarely updated), we might not need to store their bundles in the `code-intel-extensions` repo.
Another idea was to add the `build` script and bundles from this repo to [sourcegraph-extensions-cloning-scripts](https://github.com/sourcegraph/sourcegraph-extensions-cloning-scripts) repo, but it somehow contradicts that repo purpose.
So we decided to proceed with a separate repo for the code intel extensions bundles.

 > Maybe we can get the latest revision number automatically via git command?
 
 I think the person who wants to update bundles should define the revision (the latest revision may have a bug, for example, and they want to build code-intel extensions on the specific 'working' revision).

**Pull request merged**